openstack_hypervisor/hooks.py

```diff
@@ -404,6 +408,40 @@ def _configure_ovn_external_networking(snap: Snap) -> None:
         ]
     )
 
+    external_network_cidr = snap.config.get("network.external-network-cidr")
+    # Consider 0.0.0.0/0 as IPvAnyNetwork None
+    if external_network_cidr != "0.0.0.0/0":
+        logging.info(f"configuring external bridge {external_bridge}")
+
+        ipr = IPRoute()
+        dev = ipr.link_lookup(ifname=external_bridge)[0]
+        cidr = external_network_cidr.split("/")
+        try:
+            ipr.addr("add", index=dev, address=cidr[0], mask=int(cidr[1]))
+        except NetlinkError as e:
+            if not e.code == errno.EEXIST:
+                raise e
+
+        ipr.link("set", index=dev, state="up")
+
+        logging.info("updating iptables")
+        subprocess.check_call(
+            [
+                "iptables-legacy",
+                "-w",
+                "-t",
+                "nat",
+                "-A",
+                "POSTROUTING",
+                "-s",
+                external_network_cidr,
+                "-d",
+                external_network_cidr,
+                "-j",
+                "MASQUERADE",
+            ]
+        )
```

Comment:
I think it would be good to flush any prior configuration from br-ex in the event that this value is reset/unset - this could be important if a deployment moves from single node mode to multinode mode where a port to the outside world will be used instead.

Reply:
Done